src/coreclr/nativeaot/Runtime/clrgc.enabled.cpp

```diff
@@ -0,0 +1,465 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "common.h"
+#define SKIP_TRACING_DEFINITIONS
+#include "gcenv.h"
+#undef SKIP_TRACING_DEFINITIONS
+#include "gcheaputilities.h"
+#include "gchandleutilities.h"
+
+#include "gceventstatus.h"
+#include "holder.h"
+#include "RhConfig.h"
+
+enum GC_LOAD_STATUS {
+    GC_LOAD_STATUS_BEFORE_START,
+    GC_LOAD_STATUS_START,
+    GC_LOAD_STATUS_DONE_LOAD,
+    GC_LOAD_STATUS_GET_VERSIONINFO,
+    GC_LOAD_STATUS_CALL_VERSIONINFO,
+    GC_LOAD_STATUS_DONE_VERSION_CHECK,
+    GC_LOAD_STATUS_GET_INITIALIZE,
+    GC_LOAD_STATUS_LOAD_COMPLETE
+};
+
+// Load status of the GC. If GC loading fails, the value of this
+// global indicates where the failure occurred.
+GC_LOAD_STATUS g_gc_load_status = GC_LOAD_STATUS_BEFORE_START;
+
+// The version of the GC that we have loaded.
+VersionInfo g_gc_version_info;
+
+class MyGCToEEInterface : public IGCToCLR {
```

Comment:
We can use the same pattern as regular CoreCLR here. The lib split should not be a problem.

The current pattern used by regular CoreCLR has slightly better speed at the cost of duplicating native code. The current pattern used in this PR has slightly better code size, it does not duplicate native code but introduces extra method layer on GCtoEE calls. The difference is probably in the noise range.

If you like what you have in this PR better, we should unify on that regular CoreCLR. It can look like this: https://github.com/jkotas/runtime/commit/552e41bb8a055566394486e93da0cd8c0c3af8b0 . Saves ~200 lines of the boilerplate code.